dask/dataframe/core.py

```diff
@@ -6098,7 +6098,7 @@ def mapseries(base_chunk, concat_map):
 
 def mapseries_combine(index, concat_result):
     final_series = concat_result.sort_index()
-    final_series.index = index
+    final_series = pd.Series(index, index=index).map(final_series)
```

Comment:
We've been trying to avoid using `pd.Series` explicitly, in order to improve support for other dataframe libraries that are similar enough to pandas (notably RAPIDS cudf).  Can you think of a way that would produce a Series object that don't explicitly use the pandas module?

Reply:
Updated. Using the API level `take` and `argsort` methods.

Reply:
Updated -- Uses only API methods now (`take`, `sort_index` and `argsort`)

Reply:
I'll work on this a bit and see what I can do. First few attempts somehow ended up passing tests, but functionally gave wrong results, so I'm gonna update some tests too to make this cleaner.

Reply:
Actually, just found that `index.to_series()` is nearly identical to `pd.Series(index, index=index)`, so just use this `to_series()` method and everything works!